### PR TITLE
Add Hook to compat functions

### PR DESCRIPTION
**Pull request by cooldracula**

### CODE REVIEW

**Comment by Powersource** [comment]
heey cool :) could we get a test though

**Comment by Powersource** [comment]
or maybe if we do this same pattern in other files it'd maybe not be needed

**Comment by mixmix** [comment]
@Powersource this was Zach and I pairing in this area. Assume he's not coming back to this PR, then decide how you'd like to proceed. If this PR is still needed (I think this mainly exists to make testing some things easier), then can you take it over and pull me if any help needed ?

**Comment by Powersource** [comment]
yeah i think stuff like ssb-recps-guard needs this (if it detects ssb.publish it tries to hook it)

File: compat/db.js [hunk 1]
```diff
@@ -13,21 +14,21 @@ exports.manifest = {
...
       sbot.db.get(idOrObject.id, cb)
     } else {
       sbot.db.get(idOrObject, cb)
     }
-  }
-  sbot.publish = sbot.db.publish
-  sbot.whoami = () => ({ id: sbot.id })
-  sbot.ready = () => true
+  })
+  sbot.publish = Hookable(sbot.db.publish)
```
Comment (Powersource) [inline compat/db.js:27]: sbot.db.publish is created by the publish compat plugin. i wonder if this throws if we don't load that first?

Comment (Powersource) [reply compat/db.js:27]: made a new PR with some changes, didn't want to juggle the branches around https://github.com/ssbc/ssb-db2/pull/427

**Comment by Powersource** [comment]
v8.1.0

**Pull request closed without merging**